lib/compiler-singlepass/src/compiler.rs

```diff
@@ -106,29 +106,26 @@ impl Compiler for SinglepassCompiler {
             .map(|(i, input)| {
                 let reader = wasmer_compiler::FunctionReader::new(input.module_offset, input.data);
 
-                let mut local_reader = reader.get_locals_reader()?;
-                // This local list excludes arguments.
-                let mut locals = vec![];
-                let num_locals = local_reader.get_count();
-                for _ in 0..num_locals {
-                    let (count, ty) = local_reader.read()?;
-                    for _ in 0..count {
-                        locals.push(ty);
-                    }
-                }
-
                 let mut generator = FuncGen::new(
                     module,
                     module_translation,
                     &self.config,
                     &vmoffsets,
                     &table_styles,
                     i,
-                    &locals,
                     calling_convention,
                 )
                 .map_err(to_compile_error)?;
 
+                let mut local_reader = reader.get_locals_reader()?;
+                for _ in 0..local_reader.get_count() {
+                    let (count, ty) = local_reader.read()?;
+                    // Overflowing has been already been validated by the validator.
+                    generator.feed_local(count, ty).map_err(to_compile_error)?;
```

Comment:
Don't understand this comment -- it sort-of implies that we need to `.unwrap()` here but we dont'?

Reply:
It is possible that the code hasn't been run through the validator at all. I'm also not entirely confident the limits for local counts will remain in place forever, and once its gone returning an error here seems reasonable enough.